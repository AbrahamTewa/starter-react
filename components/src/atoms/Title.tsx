// ============================================================
// Import packages
import styled from 'styled-components';

// ============================================================
// Component
function Title({ children }: Props) {
  return <H1>{children}</H1>;
}

Title.defaultProps = {
  children: undefined,
};

const H1 = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: #BF4F74;
`;

type Props = {
  children: string,
};

// ============================================================
// Exports
export default Title;
