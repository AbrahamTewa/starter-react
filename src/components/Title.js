import React from 'react';
import PropTypes from 'prop-types';

/**
 * Title of the application
 * @param {string} title
 * @returns {XML}
 * @constructor
 */
function Title({title}) {
    return <h1>{title}</h1>;
}

Title.propTypes = { title: PropTypes.string.isRequired };

export default Title;

