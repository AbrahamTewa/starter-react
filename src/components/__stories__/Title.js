/* eslint-env node */

import React from 'react';
import { storiesOf } from '@storybook/react';

import Title from '../Title';

storiesOf('Button', module)
  .add('with text', () => (
      <Title/>
  ))
  .add('with some emoji', () => (
    <Title>Emoji</Title>
  ));
