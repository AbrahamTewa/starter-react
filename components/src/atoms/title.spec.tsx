/* eslint-env node, jest */

import renderer from 'react-test-renderer';

import Title from './Title';

describe('Basic tests', () => {
  test('Empty string', () => {
    const title = <Title>Hello world</Title>;

    // Snapshot
    const render = renderer.create(title);
    expect(render.toJSON()).toMatchSnapshot();
  });

  test('Simple string', () => {
    const titleText = 'One simple string';
    const title = <Title>{titleText}</Title>;

    // Snapshot
    const render = renderer.create(title);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
