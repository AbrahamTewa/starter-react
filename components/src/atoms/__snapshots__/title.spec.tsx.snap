// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Empty string 1`] = `
<h1
  className="sc-bdfCDU iEnzuz"
>
  Hello world
</h1>
`;

exports[`Basic tests Simple string 1`] = `
<h1
  className="sc-bdfCDU iEnzuz"
>
  One simple string
</h1>
`;
